linkis-public-enhancements/linkis-datasource/linkis-metadata-query/service/jdbc/src/main/java/org/apache/linkis/metadata/query/service/AbstractSqlConnection.java

```diff
@@ -0,0 +1,212 @@
+
+import org.apache.commons.collections.CollectionUtils;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class AbstractSqlConnection implements Closeable {
+
+  private static final Logger LOG = LoggerFactory.getLogger(AbstractSqlConnection.class);
+
+  public Connection conn;
+
+  public ConnectMessage connectMessage;
+
+  public AbstractSqlConnection(
+      String host,
+      Integer port,
+      String username,
+      String password,
+      String database,
+      Map<String, Object> extraParams)
+      throws ClassNotFoundException, SQLException {
+    connectMessage = new ConnectMessage(host, port, username, password, extraParams);
+    conn = getDBConnection(connectMessage, database);
+    // Try to create statement
+    Statement statement = conn.createStatement();
+    statement.close();
+  }
+
+  public abstract Connection getDBConnection(ConnectMessage connectMessage, String database)
+      throws ClassNotFoundException, SQLException;
+
+  public List<MetaColumnInfo> getColumns(String schemaname, String table)
+      throws SQLException, ClassNotFoundException {
+    List<MetaColumnInfo> columns = new ArrayList<>();
+    String columnSql = "SELECT * FROM " + schemaname + "." + table + " WHERE 1 = 2";
+    PreparedStatement ps = null;
+    ResultSet rs = null;
+    ResultSetMetaData meta;
+    try {
+      List<String> primaryKeys = getPrimaryKeys(table);
+      ps = conn.prepareStatement(columnSql);
+      rs = ps.executeQuery();
+      meta = rs.getMetaData();
+      int columnCount = meta.getColumnCount();
+      for (int i = 1; i < columnCount + 1; i++) {
+        MetaColumnInfo info = new MetaColumnInfo();
+        info.setIndex(i);
+        info.setName(meta.getColumnName(i));
+        info.setType(meta.getColumnTypeName(i));
+        if (primaryKeys.contains(meta.getColumnName(i))) {
+          info.setPrimaryKey(true);
+        }
+        columns.add(info);
+      }
+    } finally {
+      closeResource(null, ps, rs);
+    }
+    return columns;
+  }
+
+  /**
+   * Get primary keys // * @param connection connection
+   *
+   * @param table table name
+   * @return
+   * @throws SQLException
+   */
+  public List<String> getPrimaryKeys(String table) throws SQLException {
+    ResultSet rs = null;
+    List<String> primaryKeys = new ArrayList<>();
+    try {
+      DatabaseMetaData dbMeta = conn.getMetaData();
+      rs = dbMeta.getPrimaryKeys(null, null, table);
+      while (rs.next()) {
+        primaryKeys.add(rs.getString("column_name"));
+      }
+      return primaryKeys;
+    } finally {
+      if (null != rs) {
+        rs.close();
+      }
+    }
+  }
+
+  public GenerateSqlInfo getSparkSql(String database, String table) {
```

Comment:
It is not recommended to add Spark-related methods here, and a new interface should be added